firmware/src/hal/sgx/src/trusted/secret_store.c

```diff
@@ -48,18 +49,64 @@
 
 #define MAX_BLOB_SIZE (1024 * 1024)
 
+#define SECRET_PAYLOAD_SIZE(secret_size) (secret_size - sizeof(secret_header_t))
+#define SEST_HASH_SIZE (32)
+
 // To avoid performing dynamic memory allocation whenever we need to read a
 // secret, we use this static buffer to temporarily store the sealed blobs we
 // read from the untrusted modules.
 static uint8_t G_secrets_buffer[MAX_BLOB_SIZE];
```

Comment:
Shoudn't we allow additional space for the header here when we are using this for writing?

Reply:
That's a good question. The `blob` is still composed of the header + payload, so I think this line specifically is correct. But since we increased the size of the data to be sealed, you're probably right that in theory, we should probably increase `MAX_BLOB_SIZE`.

Looking at real data, though, it seems like this 1MB limit is already more than enough to our practical needs. What do you think?